net/can/can_callback.c

```diff
@@ -62,6 +62,9 @@ can_data_event(FAR struct net_driver_s *dev, FAR struct can_conn_s *conn,
                uint16_t flags)
 {
   int buflen = dev->d_len;
+#ifdef CONFIG_NET_TIMESTAMP
+  buflen -= sizeof(struct timeval);
+#endif
```

Comment:
since `recvlen = can_datahandler(dev, conn);` only carry the length of the CAN frame not the time struct, buflen should also only include CAN frame length for correct comparison after this